src/lib/ui/use-images-stack-loading.ts

```diff
@@ -1,48 +1,54 @@
-import { useCallback, useState } from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+const usePrevious = <T>(value: T): T | undefined => {
+  const ref = useRef<T>();
+  useEffect(() => {
+    ref.current = value;
+  });
```

Comment:
```suggestion
  useEffect(() => {
    ref.current = value;
  }, [value]);
```
Having a dependency looks logical to me but if everything works as intended, you may just resolve this conversation.